src/js/components/search/filters/timePeriod/TimePeriod.jsx

```diff
@@ -0,0 +1,145 @@
+import FiscalYear from './FiscalYear';
+
+const defaultProps = {
+    timePeriods: [
+        'FY 2016',
+        'FY 2015',
+        'FY 2014',
+        'FY 2013',
+        'FY 2012',
+        'FY 2011',
+        'FY 2010',
+        'FY 2009'
+    ]
+};
+
+const propTypes = {
+    timePeriods: React.PropTypes.array,
+    label: React.PropTypes.string
+};
+
+export default class TimePeriod extends React.Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            state: 'loading',
+            startDate: null,
+            endDate: null,
+            error: {
+                show: false,
+                header: '',
+                description: ''
+            },
+            errorDetails: ""
+        };
+    }
+
+    handleDateChange(date, dateType) {
+    // merge the new date into the file's state without affecting the other keys
+        this.setState({
+            [dateType]: moment(date)
+        }, () => {
+            this.validateDates();
+        });
+    }
+
+    validateDates() {
+        // validate that dates are provided for both fields and the end dates
+        // don't come before the start dates
+
+        const rng = Object.assign({}, this.state);
+
+        const output = {
+            range: rng
+        };
+
+        // validate the date ranges
+        const start = this.state.startDate;
+        const end = this.state.endDate;
+        if (start && end) {
+            // both sets of dates exist
+            if (!end.isSameOrAfter(start)) {
+                // end date comes before start date, invalid
+                // show an error message
+                output.error = {
+                    show: true,
+                    header: 'Invalid Dates',
+                    description: 'The end date cannot be earlier than the start date.'
+                };
+            }
+            else {
+                // valid!
+                output.error = {
+                    show: false,
+                    header: '',
+                    description: ''
+                };
+            }
+        }
+        else {
+            // not all dates exist yet
+            output.error = {
+                show: false,
+                header: '',
+                description: ''
+            };
+        }
+
+        this.setState(output);
+    }
+
+    showError(head, desc) {
+        this.setState({
+            error: Object.assign(this.state.error, {
+                show: true,
+                header: head,
+                description: desc
+            })
+        });
+    }
+
+    hideError() {
+        this.setState({
+            error: Object.assign(this.state.error, {
+                show: false,
+                header: '',
+                description: ''
+            })
+        });
+    }
+
+
+    render() {
+        const fiscalYears = this.props.timePeriods.map((year, index) =>
+            <FiscalYear name={year} key={index} />
```

Comment:
`FiscalYear` might make more sense semantically as a `ul` list under the hood than a series of `div`s (you can, of course, put `div`s inside each list item).